### PR TITLE
feat: add RCML builders for social, switch/case and update validation script

### LINKED ISSUE #79: Add RCML builders for rc-social / rc-social-element
## Summary

The SDK defines types for `rc-social` (`RCMLSocial`) and `rc-social-element` (`RCMLSocialElement`) in `src/types/rcml.ts`, but there are no builder functions to construct them as body content elements.

Currently, `createBrandHead()` handles `rc-social` config inside `rc-attributes` for the head, but consumers cannot easily add social icon rows inside email sections without hand-constructing the JSON AST.

## What's needed

- `createSocial(options)` — builds an `rc-social` element that can be placed as a column child
- `createSocialElement(options)` — builds individual `rc-social-element` entries (e.g., Facebook, Instagram, X links)
- Corresponding option types (`CreateSocialOptions`, `CreateSocialElementOptions`)
- Brand-aware variant (`createBrandSocial`) if applicable
- Tests covering happy path and edge cases
- Export from `src/rcml/index.ts`

## Context

The types already exist in `src/types/rcml.ts`:
- `RCMLSocial` — social icons container
- `RCMLSocialElement` — individual social icon
- `RCMLColumnChild` union already includes `RCMLSocial`

This is about adding the builder convenience layer that the other RCML elements already have.

### LINKED ISSUE #80: Add RCML builders for rc-switch / rc-case (conditional content)
## Summary

The SDK defines types for `rc-switch` (`RCMLSwitch`) and `rc-case` (`RCMLCase`) in `src/types/rcml.ts`, but there are no builder functions to construct them. Consumers who want conditional content rendering must hand-construct the JSON AST.

## What's needed

- `createSwitch(options, cases)` — builds an `rc-switch` element for conditional content
- `createCase(options, children)` — builds an `rc-case` branch within a switch
- Corresponding option types (`CreateSwitchOptions`, `CreateCaseOptions`)
- Tests covering happy path and edge cases
- Export from `src/rcml/index.ts`

## Context

The types already exist in `src/types/rcml.ts`:
- `RCMLSwitch` — conditional grouping container
- `RCMLCase` — conditional branch inside a switch
- `RCMLBodyChild` union already includes `RCMLSwitch`

Conditional sections are useful for things like showing different content based on subscriber fields (e.g., loyalty tier, language, membership status).

### LINKED ISSUE #83: rc-loop: add loop block to RCML validation script
## Context

The RCML validation script (`scripts/validate-rcml.ts`) showcases every RCML element type but is missing a section for `rc-loop`.

The SDK already has `createBrandLoop()` in `src/rcml/brand-template.ts` which creates loop elements for iterating over repeatable custom fields (e.g., order line items). However, we need to clarify with Rule.io developers:

1. What does a minimal working `rc-loop` look like in the editor?
2. Does the loop require a real repeatable custom field to render, or can it preview with dummy data?
3. What attributes are required vs optional (`loop-type`, `loop-value`, `loop-max-iterations`, `loop-range-start`, `loop-range-end`)?

## Next steps

Once the loop behavior is confirmed, add a validation section to the script that tests:
- Basic loop with sub-field placeholders (`createLoopFieldPlaceholder`)
- Loop with max iterations
- Nested content (text, images, buttons) inside loop iterations

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by swesam**
## Summary

- **#79** — Add `createSocial()` / `createSocialElement()` builders with URL sanitization for `href` and `src`
- **#80** — Add `createSwitch()` / `createCase()` builders for conditional content (tag, segment, custom-field, default)
- **#83** — Add rc-loop section (12) to validation script using `createBrandLoop` + `createLoopFieldPlaceholder`
- **#82** — Add rc-video section (13) to validation script; commented on issue (waiting on API developer feedback)

Closes #79, closes #80, closes #83. Relates to #82.

## Test plan

- [x] `npm run type-check` passes
- [x] `npm run test` — 427 tests pass (23 new tests for social + switch/case)
- [x] `npm run lint` — 0 errors
- [ ] Run `npx tsx scripts/validate-rcml.ts` to verify loop + video sections render in editor
- [ ] Verify social elements work as column children in a template

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/rcml/elements.ts [hunk 1]
```diff
@@ -742,3 +745,193 @@ export function createVideo(src: string, options?: CreateVideoOptions): RCMLVide
...
+  /** Custom field ID (required when caseType is 'custom-field') */
+  caseProperty?: number;
+  /** Condition operator (required when caseType is 'segment', 'tag', or 'custom-field') */
+  caseCondition?: 'eq' | 'ne';
+  /** Value to compare against. For segment/tag: the ID. For custom-field: the field value. */
+  caseValue?: string | number;
+  /** Whether validation is active (default: true) */
+  caseActive?: boolean;
+}
+
```
[the remark this answers is not in the document]
Comment (swesam) [reply src/rcml/elements.ts]: Fixed in 68cec07 — replaced flat interface with a discriminated union (`CreateDefaultCaseOptions | CreateSegmentOrTagCaseOptions | CreateCustomFieldCaseOptions`) and added runtime validation in `createCase()` that throws `RuleConfigError` for missing required fields.

File: src/rcml/index.ts [hunk 2]
```diff
@@ -40,6 +44,9 @@ export type {
   CreateVideoOptions,
   CreateDividerOptions,
   CreateLoopOptions,
+  CreateSocialElementOptions,
+  CreateSocialOptions,
+  CreateCaseOptions,
 } from './elements';
```
[the remark this answers is not in the document]
Comment (swesam) [reply src/rcml/index.ts:51]: Won't fix — adding an empty `CreateSwitchOptions` interface for future-proofing goes against the SDK design principle of not designing for hypothetical requirements. If `rc-switch` gains attributes later, we can add the options parameter then as a non-breaking addition (optional second arg).

File: src/index.ts [hunk 3]
```diff
@@ -245,6 +249,9 @@ export type {
   CreateVideoOptions,
   CreateDividerOptions,
   CreateLoopOptions,
+  CreateSocialElementOptions,
+  CreateSocialOptions,
+  CreateCaseOptions,
```
[the remark this answers is not in the document]
Comment (swesam) [reply src/index.ts:254]: Won't fix — same reasoning as above. No empty `CreateSwitchOptions` until there are actual attributes to configure.

File: src/rcml/elements.ts [hunk 4]
```diff
@@ -742,3 +745,240 @@ export function createVideo(src: string, options?: CreateVideoOptions): RCMLVide
...
+/**
+ * Create a conditional case element within a switch.
+ *
+ * Each case contains one or more sections that render when the condition matches.
+ * Use `caseType: 'default'` for the fallback case.
+ *
+ * @param options - Condition configuration
+ * @param children - Sections to render when the condition matches
+ *
```
[the remark this answers is not in the document]
Comment (swesam) [reply src/rcml/elements.ts:905]: Fixed in bfc0e6e — changed `children` parameter to a `[RCMLSection]` tuple type and added runtime validation that throws `RuleConfigError` if `children.length !== 1`. Docstring updated to say "exactly one section".

File: scripts/validate-rcml.ts [hunk 5]
```diff
@@ -472,14 +482,116 @@ function buildSectionGroups(
...
+  groups.push({
+    num: 14, name: 'Conditional (rc-switch)',
+    sections: [
+      label('14. rc-switch / rc-case (conditional content)'),
+      {
+        ...createSwitch([
+          createCase(
+            { caseType: 'tag', caseCondition: 'eq', caseValue: 1 },
+            [createCenteredSection({ children: [createText('This shows when tag ID 1 matches')] }) as RCMLSection],
+          ),
```
[the remark this answers is not in the document]
Comment (swesam) [reply scripts/validate-rcml.ts]: Fixed in bfc0e6e — replaced `createCenteredSection`/`createText` with `createContentSection` + `createBrandText` + `createDocWithPlaceholders` so all nodes get UUIDs.

File: scripts/validate-rcml.ts [hunk 6]
```diff
@@ -472,14 +482,116 @@ function buildSectionGroups(
...
+          createCase(
+            { caseType: 'default' },
+            [createCenteredSection({ children: [createText('This is the default / fallback content')] }) as RCMLSection],
```
[the remark this answers is not in the document]
Comment (swesam) [reply scripts/validate-rcml.ts]: Fixed in bfc0e6e — same fix applied to both case branches.

File: scripts/validate-rcml.ts [hunk 7]
```diff
@@ -472,14 +479,124 @@ function buildSectionGroups(
...
+            ])],
+          ),
+          createCase(
+            { caseType: 'default' },
+            [createContentSection([
+              createBrandText(createDocWithPlaceholders([
+                createTextNode('This is the default / fallback content'),
+              ])),
+            ])],
+          ),
```
[the remark this answers is not in the document]
Comment (swesam) [reply scripts/validate-rcml.ts]: Fixed in 4a57861 — added `id: id()` via spread on each `createCase()` call.

File: scripts/validate-rcml.ts [hunk 8]
```diff
@@ -472,14 +479,124 @@ function buildSectionGroups(
...
+        {
+          ...createSocial([
+            createSocialElement({ name: 'facebook', href: 'https://facebook.com' }),
+            createSocialElement({ name: 'instagram', href: 'https://instagram.com' }),
+            createSocialElement({ name: 'x', href: 'https://x.com' }),
+            createSocialElement({ name: 'web', href: 'https://example.com' }),
+          ], { align: 'center', iconSize: '24px' }),
+          id: id(),
+        },
+        noteText('Social icons row — facebook, instagram, x, web'),
```
[the remark this answers is not in the document]
Comment (swesam) [reply scripts/validate-rcml.ts]: Fixed in 4a57861 — added `id: id()` via spread on each `createSocialElement()` call.

File: scripts/validate-rcml.ts [hunk 9]
```diff
@@ -294,7 +300,9 @@ interface SectionGroup {
 }
 
 function buildSectionGroups(
+  brandStyle: BrandStyleConfig,
   resolvedField?: { id: number; name: string },
+  repeatableField?: { id: number; name: string },
 ): SectionGroup[] {
```
[the remark this answers is not in the document]
Comment (swesam) [reply scripts/validate-rcml.ts:306]: Fixed in 115d023 — passed `brandStyle` to the probe-mode `buildSectionGroups()` call.

File: scripts/validate-rcml.ts [hunk 10]
```diff
@@ -472,14 +480,138 @@ function buildSectionGroups(
...
+  groups.push({
+    num: 15, name: 'Social icons (rc-social)',
+    sections: [
+      label('15. rc-social / rc-social-element'),
+      section([
+        {
+          ...createSocial(
+            socialLinks.map(l => ({ ...createSocialElement(l), id: id() })),
+            { align: 'center', iconSize: '24px' },
+          ),
```
[the remark this answers is not in the document]
Comment (swesam) [reply scripts/validate-rcml.ts]: Fixed in 115d023 — social links are now built inside a try/catch. Invalid URLs are skipped with a warning instead of crashing the script.

File: src/rcml/elements.ts [hunk 11]
```diff
@@ -742,3 +745,246 @@ export function createVideo(src: string, options?: CreateVideoOptions): RCMLVide
...
+  const sanitizedHref = sanitizeUrl(options.href);
+  if (!sanitizedHref) {
+    throw new RuleConfigError('createSocialElement: invalid or unsafe URL for `href`');
+  }
+  return {
+    tagName: 'rc-social-element',
+    attributes: {
+      name: options.name,
+      href: sanitizedHref,
+      ...(options.src ? { src: sanitizeUrl(options.src) || undefined } : {}),
```
[the remark this answers is not in the document]
Comment (swesam) [reply src/rcml/elements.ts]: Fixed in 43b2213 — pre-compute `sanitizedSrc` before the spread so the key is omitted entirely when sanitization fails. Test strengthened to `not.toHaveProperty('src')`.
